Add option for specifying specific application for input run

lindsayad opened the pull request:
Closes #20062

lindsayad committed:
Add option for specifying specific application for input run

Closes #20062

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/934676/) on e96e4e9 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/20066/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/934686/) on e96e4e9 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>18c73d</th>
      <th colspan=4>#20066 e96e4e</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>82.29%</td>
      <td>82.30%</td>
      <td>+0.00%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>73194</td>
      <td>73206</td>
      <td>+12</td>
      <td>11</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15748</td>
      <td>15746</td>
      <td>-2</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20066/coverage_diff/framework.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20066/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/tensor_mechanics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/20066/coverage/xfem)

</details>



This comment will be updated on new commits.

loganharbour reviewed (dismissed):
While it is a draft - looks good to me

lindsayad committed:
Remove Registry label check

I personally have never run across this error in any real user
experiences, so I question whether it has any real value. Moreover this
error check prevents executing a specific application of a "combined"
application like blue_crab because the blue_crab executable contains
registry entries for all the objects compiled into the executable (or
more appropriately all the objects in the executable with register calls
in their .C file) yet we only `registerAll` for the specific application
we are executing. So if specifically running Griffin for example, we get
tons of errors about labels/applications: Bison, SAM, Pronghorn, etc.

lindsayad commented:
@loganharbour this is ready to review. If you approve, I'll wait to merge until we've worked out the `next` kerfuffle
